File: packages/stronghold_core/lib/models/responses/membership_payment_response.dart
import '../common/date_time_utils.dart';

/// Membership payment response from backend
class MembershipPaymentResponse {
  final int id;
  final int membershipPackageId;
  final String packageName;
  final double amountPaid;
  final DateTime paymentDate;
  final DateTime startDate;
  final DateTime endDate;

  const MembershipPaymentResponse({
    required this.id,
    required this.membershipPackageId,
    required this.packageName,
    required this.amountPaid,
    required this.paymentDate,
    required this.startDate,
    required this.endDate,
  });

  factory MembershipPaymentResponse.fromJson(Map<String, dynamic> json) {
    return MembershipPaymentResponse(
      id: json['id'] as int,
      membershipPackageId: json['membershipPackageId'] as int,
      packageName: json['packageName'] as String,
      amountPaid: (json['amountPaid'] as num).toDouble(),
      paymentDate: DateTimeUtils.parseApiDateTime(
        json['paymentDate'] as String,
      ),
      startDate: DateTimeUtils.parseApiDateTime(json['startDate'] as String),
      endDate: DateTimeUtils.parseApiDateTime(json['endDate'] as String),
    );
  }

  bool get isActive => DateTime.now().isBefore(endDate);
}

File: Stronghold/Stronghold.Application/Features/Memberships/DTOs/MembershipPaymentResponse.cs
namespace Stronghold.Application.Features.Memberships.DTOs;

public class MembershipPaymentResponse
{
    public int Id { get; set; }
    public int MembershipPackageId { get; set; }
    public string PackageName { get; set; } = string.Empty;
    public decimal AmountPaid { get; set; }
    public DateTime PaymentDate { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
}
